app/src/main/java/org/mozilla/fenix/home/intent/DeepLinkIntentProcessor.kt

```diff
@@ -76,6 +88,44 @@ class DeepLinkIntentProcessor(
                     )
                 }
             }
+            "settings_notifications" -> {
+                val intent = notificationSettings(activity)
+                activity.startActivity(intent)
+            }
+            "install_search_widget" -> {
+                if (SDK_INT >= Build.VERSION_CODES.O) {
+                    SearchWidgetCreator.createSearchWidget(activity)
+                }
+            }
         }
     }
+
+    private fun notificationSettings(context: Context, channel: String? = null) =
+        Intent().apply {
+            when {
+                SDK_INT >= Build.VERSION_CODES.O -> {
+                    action = channel?.let {
+                        putExtra(Settings.EXTRA_CHANNEL_ID, it)
+                        Settings.ACTION_CHANNEL_NOTIFICATION_SETTINGS
+                    } ?: Settings.ACTION_APP_NOTIFICATION_SETTINGS
+                    putExtra(Settings.EXTRA_APP_PACKAGE, context.packageName)
+                }
+                SDK_INT >= Build.VERSION_CODES.LOLLIPOP -> {
+                    action = "android.settings.APP_NOTIFICATION_SETTINGS"
+                    putExtra("app_package", context.packageName)
+                    putExtra("app_uid", context.applicationInfo.uid)
+                }
+                else -> {
+                    action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+                    addCategory(Intent.CATEGORY_DEFAULT)
+                    data = Uri.parse("package:" + context.packageName)
+                }
+            }
+        }
+        .apply {
+            when {
+                SDK_INT >= Build.VERSION_CODES.P ->
```

Comment:
why does just P and above need this ?

Reply:
Hmm. This was a fairly mechanical re-write from [this question][1]. I've removed it, because it doesn't break anything.

[1]: https://stackoverflow.com/questions/32366649/any-way-to-link-to-the-android-notification-settings-for-my-app